### ISSUE TITLE
Spurious links generated for C++ keywords and template parameters

### ISSUE THREAD

**Issue by marktsuchida** [doxygen/doxygen#10797]
Under conditions that I don't fully understand, things that shouldn't be links (such as the C++ keywords `auto`, `void`, `template`, and template parameter names) become links to some (seemingly randomly chosen, but reproducible) documentation page. The spurious links occur in both brief and detailed function and typedef docs, among other places. Minimal repro below.

**Screenshot**
![image](https://github.com/doxygen/doxygen/assets/660808/2a7c20b7-197b-4483-a43e-de4fabc6dada)
In this example (the reproduction attached below), the `void` keyword on member function `a()` is a link to the page for `namespace myproject`. This `void` keyword is also contained in a `ref` element in the XML output.

In other cases, the link target was sometimes a topic page or a particular function. It looked like all of the spurious links in the project went to the same page or anchor, but the target did change several times as I was minimizing the example.

**To Reproduce**
Run `doxygen` in the directory extracted from [repro.tgz](https://github.com/doxygen/doxygen/files/14969030/repro.tgz).
I minimized this from a medium-sized project in which all sorts of items turn into links to a particular page.

**Expected behavior**
These keywords and template parameters should not be links at all.

**Version**
All of the following reproduced the bug:
`master` (eb110639dcc32804b59abdf0dcb343abde211b1e), self-built, macOS arm64.
1.10.0, from Homebrew, macOS arm64.
1.9.8 (c2fe5c3e4986974eb2a97608b24086683502f07f), macOS arm64, from the GitHub Releases .dmg (using the embedded `doxygen` from the command line).

**Comment by albert-github** [comment]
Looks like a regression on (found by bisection):
```
Commit: 9a8b3dad55672e36f59c04bbfcb29621394c5e5a [9a8b3da]

Date: Sunday, June 18, 2023 8:15:42 PM

issue #7580 No autolink for template functions
```

With the issue #7580 I also see that there still looks like to be an issue.

**Comment by doxygen** [comment]
@marktsuchida Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by marktsuchida** [comment]
@doxygen Yes, it does indeed fix the problem. Thank you so much for the fast fix!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by marktsuchida** [comment]
Yes, this remains fixed in 1.11.0 (built from source on macOS). Thank you!
